src/iterators/mod.rs

```diff
@@ -47,6 +47,19 @@ pub struct HistogramIterator<'a, T: 'a + Counter, P: PickyIterator<T>> {
     picker: P,
 }
 
+/// The value emitted at each step when iterating over a `Histogram`.
+#[derive(Debug, PartialEq)]
+pub struct IterationValue<T: Counter> {
+    /// the lowest value stored in the current histogram bin
+    pub value: u64,
+    /// percent of recorded values that are equivalent to or below `value`
+    pub percentile: f64,
+    /// recorded count for values equivalent to `value`
+    pub count_at_value: T,
+    /// number of values traversed since the last iteration step
+    pub count_since_last_iteration: u64
+}
+
```

Comment:
One possibility would be to use something like https://play.rust-lang.org/?code=%0Ause%20immutable%3A%3AImmutable%3B%0A%0Astruct%20Foo%20%7B%20bar%3A%20u64%2C%20baz%3A%20u64%20%7D%0A%0Aimpl%20Foo%20%7B%0A%20%20%20%20fn%20sum(%26self)%20-%3E%20u64%20%7B%0A%20%20%20%20%20%20%20%20self.bar%20%2B%20self.baz%0A%20%20%20%20%7D%0A%20%20%20%20%0A%20%20%20%20fn%20reset(%26mut%20self)%20%7B%0A%20%20%20%20%20%20%20%20self.bar%20%3D%200%3B%0A%20%20%20%20%20%20%20%20self.baz%20%3D%200%3B%0A%20%20%20%20%7D%0A%7D%0A%0Afn%20get_foo()%20-%3E%20Immutable%3CFoo%3E%20%7B%0A%20%20%20%20Immutable%3A%3Anew(Foo%20%7B%20bar%3A%201%2C%20baz%3A%201%20%7D)%0A%7D%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20x%20%3D%20get_foo()%3B%0A%20%20%20%20%0A%20%20%20%20println!(%22%7B%7D%20%2B%20%7B%7D%20%3D%20%7B%7D%22%2C%20x.bar%2C%20x.baz%2C%20x.sum())%3B%0A%20%20%20%20%0A%20%20%20%20%2F%2F%20x.reset()%3B%20%2F%2F%20error%3A%20cannot%20borrow%20immutable%20borrowed%20content%20as%20mutable%0A%20%20%20%20%2F%2F%20x.bar%20%3D%205%3B%20%2F%2F%20error%3A%20cannot%20assign%20to%20immutable%20field%0A%7D%0A%0Amod%20immutable%20%7B%0A%20%20%20%20use%20std%3A%3Aops%3A%3ADeref%3B%0A%20%20%20%20%0A%20%20%20%20pub%20struct%20Immutable%3CT%3E(T)%3B%0A%20%20%20%20%0A%20%20%20%20impl%3CT%3E%20Immutable%3CT%3E%20%7B%0A%20%20%20%20%20%20%20%20pub%20fn%20new(t%3A%20T)%20-%3E%20Self%20%7B%20Immutable(t)%20%7D%0A%20%20%20%20%20%20%20%20pub%20fn%20get(%26self)%20-%3E%20%26T%20%7B%20%26self.0%20%7D%0A%20%20%20%20%7D%0A%20%20%20%20%0A%20%20%20%20impl%3CT%3E%20Deref%20for%20Immutable%3CT%3E%20%7B%0A%20%20%20%20%20%20%20%20type%20Target%20%3D%20T%3B%0A%20%20%20%20%20%20%20%20fn%20deref(%26self)%20-%3E%20%26T%20%7B%20%26self.0%20%7D%0A%20%20%20%20%7D%0A%7D%0A&version=stable&backtrace=0.